llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -758,11 +759,21 @@ transform_ext::StructuredOpMatcher::singleOpWithCanonicaleArgs(
       return false;
     if (yieldOp->getOperand(0).getDefiningOp() != innerOp)
       return false;
-    for (auto [index, operand] : llvm::enumerate(innerOp->getOperands())) {
-      auto arg = dyn_cast<BlockArgument>(operand);
-      if (!arg || arg.getParentBlock() != linalgOp.getBlock() ||
-          arg.getArgNumber() != index)
+    if (commutative && innerOp->getNumOperands() == 2) {
+      auto arg0 = dyn_cast<BlockArgument>(innerOp->getOperand(0));
+      auto arg1 = dyn_cast<BlockArgument>(innerOp->getOperand(1));
+      if (!arg0 || !arg1)
+        return false;
+      if (!((arg0.getArgNumber() == 0 && arg1.getArgNumber() == 1) ||
```

Comment:
This no longer checks whether `arg0` and `arg1` are arguments of the `linalgOp` block, and will catch, e.g., function arguments.

Reply:
thanks, added back